src/hotspot/cpu/x86/x86.ad

```diff
@@ -7313,6 +7419,53 @@ instruct vucast(vec dst, vec src) %{
   ins_pipe( pipe_slow );
 %}
 
+// Convert from Halffloat to other types
+instruct vcvtHFtoD_reg(vec dst, vec src) %{
+  predicate(UseAVX > 2 && VM_Version::supports_avx512_fp16() && Matcher::vector_element_basic_type(n) == T_DOUBLE);
```

Comment:
Should there be a reverse check !FP16 in vcastStoX_evex to resolve any dis-ambiguity during selection

Reply:
Being a common IR, we need to add a complimentary check in the short vector casting.